src/test/kotlin/com/hibob/academy/service/PetServiceTest.kt

```diff
@@ -66,4 +71,40 @@ class PetServiceTest {
         }
         assertEquals("No owner with ID $petId", exception.message)
     }
+
+    @Test
+    fun `test getOwnerPets returns list of pets`() {
+        val ownerId = 101L
+        val pets = listOf(
+            Pet(id = 1L, name = "Rex1", type = PetType.DOG, companyId = 1L, dateOfArrival = LocalDate.of(2024, 1, 6), ownerId = ownerId),
+            Pet(id = 2L, name = "Rex2", type = PetType.CAT, companyId = 1L, dateOfArrival = LocalDate.of(2022, 2, 5), ownerId = ownerId),
+            Pet(id = 3L, name = "Rex3", type = PetType.DOG, companyId = 1L, dateOfArrival = LocalDate.of(2026, 3, 5), ownerId = ownerId)
+        )
+
+        whenever(petDao.getOwnerPets(eq(ownerId))).thenReturn(pets)
+
+        val result = petService.getOwnerPets(ownerId)
+
+        assertEquals(pets, result)
+    }
+
+    @Test
+    fun `test countPetsByType returns correct pet counts`() {
```

Comment:
I'm using another func from petDao (getPetsByType) to adhere to DRY principle, so I didn't create a new func inside petDao for counting.
It seemed close enough for me.